src/octopus-cli-wrapper.ts

```diff
@@ -0,0 +1,180 @@
+import { InputParameters } from './input-parameters'
+import { info, setFailed } from '@actions/core'
+import { exec, ExecOptions } from '@actions/exec'
+
+// environment variables can either be a NodeJS.ProcessEnv or a plain old object with string keys/values for testing
+type EnvVars = { [key: string]: string } | NodeJS.ProcessEnv
+
+export class OctopusCliWrapper {
+  inputParameters: InputParameters
+  env: EnvVars
+  logInfo: (message: string) => void
+  logWarn: (message: string) => void
+
+  constructor(
+    parameters: InputParameters,
+    env: EnvVars,
+    logInfo: (message: string) => void,
+    logWarn: (message: string) => void
+  ) {
+    this.inputParameters = parameters
+    this.env = env
+    this.logInfo = logInfo
+    this.logWarn = logWarn
+  }
+
+  stdLine(line: string): void {
```

Comment:
I used `stdline` with a lowercase L to match execOptions. I don't mind if we have capital L here but we should be consistent across projects